lifestuff/src/http_utils.rs

```diff
@@ -51,9 +55,37 @@ pub(crate) fn normalize_api_url(base_url: String) -> String {
     base_url
 }
 
+/// Get the API key from environment variable
+pub(crate) fn get_api_key() -> Option<String> {
+    std::env::var(API_KEY_ENV_VAR)
+        .ok()
+        .filter(|k| !k.is_empty())
+}
+
+/// Build default headers for API requests, including API key if available.
+/// For non-localhost endpoints, warns if API key is missing.
+pub(crate) fn build_request_headers(base_url: &str) -> HeaderMap {
+    let mut headers = HeaderMap::new();
+    headers.insert(CONTENT_TYPE, HeaderValue::from_static("application/json"));
+
+    if let Some(api_key) = get_api_key() {
+        if let Ok(value) = HeaderValue::from_str(&api_key) {
```

Comment:
I recommend you also do this https://docs.rs/reqwest/latest/reqwest/header/struct.HeaderValue.html#method.set_sensitive